homeassistant/components/unifiprotect/binary_sensor.py

```diff
@@ -0,0 +1,271 @@
+"""This component provides binary sensors for UniFi Protect."""
+from __future__ import annotations
+
+import asyncio
+from copy import copy
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+from typing import TYPE_CHECKING, Any, Final
+
+from pyunifiprotect.data import NVR, Camera, Light, Sensor
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_LAST_TRIP_TIME, ATTR_MODEL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN, RING_INTERVAL
+from .data import ProtectData
+from .entity import ProtectDeviceEntity, ProtectNVREntity, async_all_device_entities
+from .models import ProtectRequiredKeysMixin
+from .utils import get_nested_attr
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# Remove when 3.8 support is dropped
+if TYPE_CHECKING:
```

Comment:
If this is going to be needed another models, can you put it in cost?

Reply:
I do not think it is needed elseware, and if I refactor to use `async_call_later`, I think I can get rid of that.